### ISSUE TITLE
suggestion for docs: collapse markdown link URLs

### ISSUE THREAD

**Issue by tecunningham** [moalamri/vscode-inline-fold#70]
I didn't see this in the documentation, but the extension is very useful for collapsing markdown link URLs:

Regex: `\[.*\]\((.*)\)`
Regex Flags: `g`

<img width="156" alt="image" src="https://user-images.githubusercontent.com/14047958/202469572-c46fd851-6721-46a1-801c-2d4ab576b436.png">

**Comment by moalamri** [comment]
Hi @tecunningham 
Users are allowed to document where the extension can help in different cases.
I will be happy to merge any use case you can suggest such this one 👍 it looks really useful

**Comment by tecunningham** [comment]
OK cool I will play around with it a bit more. The one downside is that when I have word-wrap on, and the URL is long, then the blank lines get rendered. I think there's no workaround for that right?

<img width="321" alt="image" src="https://user-images.githubusercontent.com/14047958/202760841-2f70e95a-c257-4586-b2a6-36d0d9bc4497.png">

**Comment by moalamri** [comment]
@tecunningham Yeah we noticed that, unfortunately the only workaround is to disable the word-wrap 😑.
Regarding any code formatting where it will actually break lines, I've thought of hooking a code editor with the extension to force some prettifying rules. But that will be an overhead since it must be language aware.
I must add limitations note in the home page until VS Code support horizontal folding.

**Comment by ctf0** [comment]
@moalamri , can we have support for regex per language as well ?, instead of one regex fits all solution

**Comment by e-medeiros** [comment]
@tecunningham added #75

**Comment by tecunningham** [comment]
Note that this is an improvement: it makes the regexp non-greedy, so will match the first closing parantheses (")") that it finds:

"inlineFold.regex": "\\[.*\\]\\(([^\\)]*)\\)",

E.g. suppose you have this:

```
[one](one) (asdfjkl)
```

Then the previous regexp will fold *everything* up to the last parantheses.

**Comment by moalamri** [comment]
Hey @tecunningham 
Urls can contain parentheses as well, I guess being greedy is the better option. That being said we can tweak the Regex to consider the ending valid if is ")" followed by a space "\s", that way we can at least narrow down mismatches?

**Comment by tecunningham** [comment]
Oh good point. OK working on an improved regexp, and I think I've solved the "multiple matches on one line" issue too. Will update later today.

**Comment by tecunningham** [comment]
OK I suggest this as new recommendation for markdown links:

```
   "inlineFold.regexGroup": "2",
   "inlineFold.regex": "\\[([\\w\\s\\d]+)\\]\\(([^\\s]+)\\)[,.!\\?]*\\s",
   "inlineFold.regexFlags": "g",
```

Here are a bunch of test cases:

```
**Should fold:**

- [filename link](../abc.md)

- [filename link with parantheses](../Twitter/Twitter_Private%202022-10-29,%2008_53%20PM%20(22%20items).csv)

- [HTTP link](http://asdf.com)

- [HTTPS link](https://en.m.wikipedia.org/wiki/Wikipedia:Wikipedia_Signpost/2022-10-31/News_and_notes#abc)

- [HTTP link with parentheses](http://asdfj.com/asdf(asdf))

- [HTTP link with punctuation afterwards](http://google.com)...! 

- Two HTTP links on one line: [one](http://google.com/), [two](http://google.com/)

- Three HTTP links on one line: [one](http://google.com/), [two](http://google.com/), [three](http://google.com/)

- Four HTTP links: [one](http://google.com/) [two](http://google.com/) [three](http://google.com/) [four](http://google.com/)

- Four relative links: [one](one) [two](two) [three](three) [four](four)


**Should not fold:**

- [HTTP with space](http://asdjfkl .com)

- [link with text straight afterwards](http://google.com/)asdf
```

<img width="493" alt="image" src="https://user-images.githubusercontent.com/14047958/215180478-c9adc4ed-2d80-4ad4-977b-eeb8dd14b35b.png">

**Comment by moalamri** [comment]
Awesome 👍

**Comment by rubenvandeven** [comment]
> ... I must add limitations note in the home page until VS Code support horizontal folding.

For those interested (as I was) this required feature is tracked in microsoft/vscode#50840.

Thanks @moalamri for working around this limitation as far as possible!
